Build for NRF52840 fails when SETTINGS_RAM=1

Hi! I faced with undefined reference to `otPlatSettingsDeinit' exception when tried to build libraries for NRF52840 platform. OpenThread commit id is 3161e9bac2a1576009d930a81558edad4c1aaceb (latest development branch)

I tried to execute the command:
```
make -f examples/Makefile-nrf52840 SETTINGS_RAM=1 BORDER_AGENT=1 BORDER_ROUTER=1 COAP=1 COMMISSIONER=1 DNS_CLIENT=1 JOINER=1 LINK_RAW=1 MAC_FILTER=1 MTD_NETDIAG=1 SERVICE=1 UDP_FORWARD=1 ECDSA=1 SNTP_CLIENT=1 COAPS=1 FULL_LOGS=1 BOOTLOADER=BLE DISABLE_BUILTIN_MBEDTLS=0 TIME_SYNC=1
```

And got next logs:

```
Making all in apps
Making all in cli
  CCLD     ot-cli-ftd
  CCLD     ot-cli-mtd
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: ../../../src/core/libopenthread-mtd.a(libopenthread_mtd_a-settings.o): in function `ot::Settings::Deinit()':
*****/openthread/examples/../src/core/common/settings.cpp:102: undefined reference to `otPlatSettingsDeinit'
collect2: error: ld returned 1 exit status
make[6]: *** [Makefile:1292: ot-cli-mtd] Error 1
make[6]: *** Waiting for unfinished jobs....
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: ../../../src/core/libopenthread-ftd.a(libopenthread_ftd_a-settings.o): in function `ot::Settings::Deinit()':
*****/openthread/examples/../src/core/common/settings.cpp:102: undefined reference to `otPlatSettingsDeinit'
collect2: error: ld returned 1 exit status
make[6]: *** [Makefile:1288: ot-cli-ftd] Error 1
make[5]: *** [Makefile:447: all-recursive] Error 1
make[4]: *** [Makefile:449: all-recursive] Error 1
make[3]: *** [Makefile:606: all-recursive] Error 1
make[2]: *** [Makefile:532: all] Error 2
```

--- a/examples/platforms/utils/settings_ram.c
+++ b/examples/platforms/utils/settings_ram.c
@@ -63,6 +63,11 @@ void otPlatSettingsInit(otInstance *aInstance)
     sSettingsBufLength = 0;
 }
 
+void otPlatSettingsDeinit(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+}
+
 otError otPlatSettingsGet(otInstance *aInstance, uint16_t aKey, int aIndex, uint8_t *aValue, uint16_t *aValueLength)
 {
     OT_UNUSED_VARIABLE(aInstance);

